go/pkg/cas/upload_test.go

```diff
@@ -105,35 +107,49 @@ func TestFS(t *testing.T) {
 	}
 }
 
-func TestChecks(t *testing.T) {
+func TestSmallBlobs(t *testing.T) {
 	t.Parallel()
 	ctx := context.Background()
 
 	var mu sync.Mutex
 	var gotDigestChecks []*repb.Digest
-	var gotRequestSizes []int
-	var gotScheduleUploadCalls []*uploadItem
+	var gotDigestCheckRequestSizes []int
+	var gotUploadBlobReqs []*repb.BatchUpdateBlobsRequest_Request
+	failCBlob := true // blob "c" is uploaded below.
 	cas := &fakeCAS{
 		findMissingBlobs: func(ctx context.Context, in *repb.FindMissingBlobsRequest, opts ...grpc.CallOption) (*repb.FindMissingBlobsResponse, error) {
 			mu.Lock()
 			defer mu.Unlock()
 			gotDigestChecks = append(gotDigestChecks, in.BlobDigests...)
-			gotRequestSizes = append(gotRequestSizes, len(in.BlobDigests))
+			gotDigestCheckRequestSizes = append(gotDigestCheckRequestSizes, len(in.BlobDigests))
 			return &repb.FindMissingBlobsResponse{MissingBlobDigests: in.BlobDigests[:1]}, nil
 		},
+		batchUpdateBlobs: func(ctx context.Context, in *repb.BatchUpdateBlobsRequest, opts ...grpc.CallOption) (*repb.BatchUpdateBlobsResponse, error) {
```

Comment:
FWIW I think we might actually want to converge this with fakeCAS at some point (no need for now) - otherwise we'll get quite a bit of repeated code as you write more tests.

Reply:
see also the previous discussion with Takuto in this PR.

Reply:
Right - doesn't seem like it's important now. Let's see if we end up having to duplicate it.